File: Maths/aires.js
/**
 * Différentes fonctions de calcul de l'aire de différentes figures.
 * (sphère, cube, cylindre, cone, tore)
 * @author Noam Zeitoun
 * @version 1.0
 */

/**
 * Calcul de l'aire d'une sphère
 * @param {number} r - Rayon de la sphère
 * @return {number} Aire de la sphère
 */
const aireSphère = (r) => 4 * Math.PI * r ** 2;

/**
 * Calcul de l'aire d'un cube
 * @param {number} c - Côté du cube
 * @return {number} Aire du cube
 */
const aireCube = (c) => 6 * c ** 2;

/**
 * Calcul de l'aire d'un cylindre
 * @param {number} r - Rayon du cylindre
 * @param {number} h - Hauteur du cylindre
 * @return {number} Aire du cylindre
 */
const aireCylindre = (r, h) => 2 * Math.PI * r * h + 2 * Math.PI * r ** 2;

/**
 * Calcul de l'aire d'un cone
 * @param {number} r - Rayon du cone
 * @param {number} h - Hauteur du cone
 * @return {number} Aire du cone
 */
const aireCone = (r, h) => Math.PI * r * (r + Math.sqrt(r ** 2 + h ** 2));

/**
 * Calcul de l'aire d'un tore
 * @param {number} r - Rayon du tore
 * @param {number} R - Rayon du tube
 * @return {number} Aire du torus
 */
const aireTore = (r, R) => 4 * Math.PI ** 2 * r * R;

// Tests
console.table({
  "aireSphère(3)": aireSphère(3),
  "aireCube(3)": aireCube(3),
  "aireCylindre(3, 3)": aireCylindre(3, 3),
  "aireCone(3, 3)": aireCone(3, 3),
  "aireTore(3, 3)": aireTore(3, 3),
});

File: Python/Maths/aires.py
from math import pi, sqrt
"""
Description:
    Différentes fonctions de calcul de l'aire de différentes figures.

Auteur:
    Noam Zeitoun

Version:
    1.0
"""

def aire_sphere(r: float) -> float:
    """
    Description:
        Fonction qui calcule l'aire d'une sphère.
    
    Paramètres:
        r: {float} -- Le rayon de la sphère.
    
    Retourne:
        {float} -- L'aire de la sphère.

    Exemple:
        >>> aire_sphere(3)
        113.09733552923255
    """
    return 4 * pi * (r ** 2)


def aire_cube(c: float) -> float:
    """
    Description:
        Fonction qui calcule l'aire d'un cube.
    
    Paramètres:
        c: {float} -- La longueur d'un côté du cube.
    
    Retourne:
        {float} -- L'aire du cube.

    Exemple:
        >>> aire_cube(3)
        54
    """
    return 6 * (c ** 2)

def aire_cylindre(r: float, h: float) -> float:
    """
    Description:
        Fonction qui calcule l'aire d'un cylindre.
    
    Paramètres:
        r: {float} -- Le rayon du cylindre.
        h: {float} -- La hauteur du cylindre.
    
    Retourne:
        {float} -- L'aire du cylindre.

    Exemple:
        >>> aire_cylindre(3, 3)
        113.09733552923255
    """
    return 2 * pi * r * h + (2 * pi * (r ** 2))

def aire_cone(r: float, h: float) -> float:
    """
    Description:
        Fonction qui calcule l'aire d'un cone.
    
    Paramètres:
        r: {float} -- Le rayon du cone.
        h: {float} -- La hauteur du cone.
    
    Retourne:
        {float} -- L'aire du cone.

    Exemple:
        >>> aire_cone(3, 3)
        68.26028032573343
    """
    return pi * r * (r + sqrt(h ** 2 + r ** 2))

def aire_tore(r: float, R: float) -> float:
    """
    Description:
        Fonction qui calcule l'aire d'un tore.
    
    Paramètres:
        r: {float} -- Le rayon du tore.
        R: {float} -- Le rayon du tube.
    
    Retourne:
        {float} -- L'aire du tore.

    Exemple:
        >>> aire_tore(3, 3)
        355.3057584392169
    """
    return 4 * pi ** 2 * r * R

# TESTS
table_data = [
    [
        'aire_sphere(3)', 
        'aire_cube(3)', 
        'aire_cylindre(3, 3)', 
        'aire_cone(3, 3)', 
        'aire_tore(3, 3)'
    ],
    [
        aire_sphere(3), 
        aire_cube(3), 
        aire_cylindre(3, 3), 
        aire_cone(3, 3), 
        aire_tore(3, 3)
    ]
]
for row in table_data:
    print("{: <20} {: <20} {: <20} {: <20} {: <20}".format(*row))